### ISSUE TITLE
Handle Symbol with longString names

### ISSUE THREAD

**Issue by nchevobbe** [devtools-html/debugger.html#6139]
_From @nchevobbe on July 25, 2017 13:7_

**Steps to reproduce:** 
Evaluate `Symbol("aa".repeat(10000))` in the Reps test app

**Expected results**
The symbol is rendered as `Symbol(aaaaaaaaaaaaaa…aaaaaaaaaaaaaa)`

**Actual results**
The symbol is rendered as `Symbol([object Object])`

---

Here is the RDP packet we receive : 
```json
{
  "type": "symbol",
  "name": {
    "type": "longString",
    "initial": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "length": 20000,
    "actor": "server2.conn1.child1/longString44"
  }
}
```

And here is the code where we handle the Symbol name : https://github.com/devtools-html/devtools-core/blob/master/packages/devtools-reps/src/reps/symbol.js#L26 

The error is due to the fact that we do a concatenation of the name 
```js
`Symbol(${name || ""})`
```
but here the name property is a longString actor.

To fix this issue, we should : 
- Test if the name is a string, and if so, do the same thing as we do
- If not, render the name as its own element using `Rep({object: name})`, which will take care of displaying the longString as we need.

_Copied from original issue: devtools-html/devtools-core#518_

**Comment by kootoopas** [comment]
@nchevobbe Hey there Nicolas.

In what way do you think it's best to handle the longString? We can't exactly depict it as you specified in the "expected results" section, since we don't have the `fullText` from the beginning and the cropping would be performed on the `initial` string:
![screen shot 2018-05-04 at 10 31 14](https://user-images.githubusercontent.com/601001/39617260-ad5dfa70-4f87-11e8-9902-a022fa2d5062.png)

We'd need to load it as soon as possible, displaying a loading text or something until it does so.

Or we could render the Symbol whose object is a longString through the `Tree`. Here's an imitiation of what that would look like:
![screen shot 2018-05-04 at 10 31 07](https://user-images.githubusercontent.com/601001/39617259-ad3b4bba-4f87-11e8-99be-d94fd6409eff.png)

**Comment by kootoopas** [comment]
Ideally we would use `getRep(object.name)`, so that we'd get valid representations for Symbols whose name is of other non-primitive types too:
![screen shot 2018-05-04 at 11 04 17](https://user-images.githubusercontent.com/601001/39618219-71cf38d0-4f8b-11e8-978c-2cb71ef9b1d3.png)

**Comment by nchevobbe** [comment]
As a first step, let's only display the truncated name, without a way to expand the longString, since I don't think it is a frequent to have a long string as a Symbol name.

**Comment by kootoopas** [comment]
Agreed. The whole thing with longStrings is kind of an edge case... but for some reason I'm obsessed with its issues. 😆
